ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/MockCrypto.hs

```diff
@@ -36,3 +40,29 @@ instance HashAlgorithm h => Crypto (TPraosMockCrypto h) where
 instance HashAlgorithm h => TPraosCrypto (TPraosMockCrypto h)
 
 type Block h = ShelleyBlock (TPraosMockCrypto h)
+
+-- | Like 'TPraosMockCrypto' except mocks fewer components in order to be
+-- compatible with bootstrapping from Byron
+--
+-- Note that mocking less components does prevent the @cardano-ledger-specs@
+-- generators from being used. Currently, this is not an obstacle for example
+-- in the Cardano ThreadNet tests.
+--
+-- NOTE: The "Ouroboros.Consensus.Cardano.CanHardFork" translation currently
+-- assumes that @ADDRHASH@ has the same bit size as Byron address hashes (ie
+-- 224); that's why we use 'Blake25_224' here.
+--
+-- NOTE: The @shelley-spec-ledger@ package currently requires that @'DSIGN' ~
+-- 'Ed25519DSIGN'.
+data TPraosMockCryptoCompatByron h
```

Comment:
Some thoughts about parameterising mock crypto by `h` (also `TPraosMockCrypto`): in #2503, I rewrite the examples used by the golden tests to use `TPraosStandardCrypto`, so no more `TPraosMockCrypto` there. I believe the serialisation roundtrip tests still use `ShortHash` for performance reasons and for generator reuse/ease (IIRC). The Shelley ThreadNet tests also use `ShortHash` for performance. Any other things I'm missing? All other things uses Mock* with `Blake2b_256`, I suppose. So actually there's no need to parameterise `TPraosMockCryptoCompatByron` over `h`, correct? Nevertheless, I think it's fine to leave it as-is.

Reply:
Yeah, `git grep -e 'TPraosMockCrypto' -- '*hs' | grep -vE -e "TPraosMockCrypto[')]" -e 'TPraosMockCrypto (h|ShortHash)'` confirms that your list is exhaustive for `TPraosMockCrypto`.

And this PR only uses the new `TPraosMockCryptoCompatByron` at type `Blake2b_256`.

But I agree the parameterization is fine as-is. Some constraint contexts would be simpler without it, but it's not burdensome to currently justify the extra diff now, in my opinion. In fact, some functions and certainly some types are clearer/safer when paramterized over `h`, in my opinion.